components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -511,35 +511,29 @@ private void handlePostConsent(OAuthMessage oAuthMessage) throws ConsentHandling
             }
 
             List<Integer> approvedClaimIds = getUserConsentClaimIds(oAuthMessage);
-            if (isPostConsentHandlingRequired(approvedClaimIds)) {
```

Comment:
Just wondering what will happen if we have already approved some of the consent. Ex: Claims A, B are requested and already approved. Then later Claim C is requested. Will it ask only for Claim C in the next authentication? By the looks of the code shouldn't we compare already approved claims and requesting claims and then decide `isPostConsentHandlingRequired ` ?

Reply:
As you said, when claim C is requested, it will only ask the consent for claim C and it will handle trough the pre consent and here it handles the post consent. With the `getUserConsentClaimIds(oAuthMessage)` method it will get the approved claim Ids to pass to the `processConsent` method[1] to process consent and store the approved and denied consents accordingly. The `isPostConsentHandlingRequired(approvedClaimIds)` condition will true only if there is an approvedClaimIds. Hence removed that condition in order to the claims that are denied also.

Let's say  Claims A, B are requested and already approved. Then later Claim C is requested and didn't give consent. Then according to the fix done at [2], in the "CM_SP_PURPOSE_PII_CAT_ASSOC" table, we store it as a consent denied claim. Hence `(isPostConsentHandlingRequired(approvedClaimIds))` with this condition that denied consent storing part will be ignored and that's the reason to remove that condition.

[1] https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1555/files#diff-debebfb0b01b553afd77f73d24e1df466030d3d19b218e0d7b3005364a05939aR535
[2] https://github.com/wso2/carbon-consent-management/pull/180